### ISSUE TITLE
[Proposal] helper array random should be better to return the array itself rather than throwing an exception when array length less than requested

### ISSUE THREAD

**Issue by mirfmaster** [laravel/ideas#2322]
It happened when I try this
`SomeModel::all()->random(3);`

if I take a look at vendor/laravel/framework/src/Illuminate/Support/Arr.php
`public static function random($array, $number = null)
    {
        $requested = is_null($number) ? 1 : $number;

        $count = count($array);

        if ($requested > $count) {
            throw new InvalidArgumentException(
                "You requested {$requested} items, but there are only {$count} items available." // here throwing exception
            );
        }

        if (is_null($number)) {
            return $array[array_rand($array)];
        }

        if ((int) $number === 0) {
            return [];
        }

        $keys = array_rand($array, $number);

        $results = [];

        foreach ((array) $keys as $key) {
            $results[] = $array[$key];
        }

        return $results;
    }`

It would be better to return the array itself rather than stopping the apps with the exception like this
`if ($requested > $count) {
            return $array;
        }`

**Comment by avido** [comment]
But you still have to write logic like 
```php
if ($returnedCount < $requestCount) { 
$apparentlyItFailed = true;
}```

Seems to me no difference then 
```php
try {
$requested = SomeModel::all()->random(3);
} catch (InvalidArgumentException $e) {
$apparentlyItFailed = true;
}```

**Comment by mirfmaster** [comment]
> But you still have to write logic like
> 
> ```
> if ($returnedCount < $requestCount) { 
> $apparentlyItFailed = true;
> }
> ```
> 
> Seems to me no difference then
> 
> ```
> try {
> $requested = SomeModel::all()->random(3);
> } catch (InvalidArgumentException $e) {
> $apparentlyItFailed = true;
> }
> ```

No, I want get random record just foreach it on view not for an expression.
wouldn't that be annoying if we write this on our controller
`try {
$requested = SomeModel::all()->random(3);
$data = Data::all();
$other = Other::all();

return view('view')->withData($data)->withOther($other)->withRequested($requested);
} catch (InvalidArgumentException $e) {
$data = Data::all();
$other = Other::all();

return view('view')->withData($data)->withOther($other)->withRequested([]);
}
`
